### PR TITLE
Set up bookSlice and   categoriesSlice

### PR DESCRIPTION
**Pull request by edahigure**
- Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- Create a directory that will contain all your Redux logic (/src/redux)
- Configure a Redux store (/src/redux/store.js)
- Define a slice of state for books that:
- Includes an array of books (initial state: empty array)
- Includes a reducer that adds a book
- Includes a reducer that removes a book
- Define a slice of state for categories that:
- Includes an array of categories (initial state: empty array)
- Includes a reducer that check the status and always return "Under construction" (the initial state should checge to that string)
- Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    removeBook: (state, action) => {
+      const { id } = action.payload;
+      state.bookList = state.bookList.filter((item) => item.id !== id);
+      state.id -= 1;
+      console.log('ID', id);
+      console.log('payload', action.payload);
+      console.log('state', state.bookList);
+    },
+  },
+});
```
Comment (brainconnect93) [inline src/redux/books/booksSlice.js]: - [ ] Hi @edahigure, great effort working on your bookstore project, but i notice in your project you have given some data passed in as `array of an object` in your bookSlice. and with this anytime i tried to add a new book to the list, it was only the ones you have in the array that got publish to the list. I can only remove the books you have in the array, but anytime i tried to add a new book, it is the ones you have in the array that got published.
  - I would advise you set the initial state of the book to an empty array. Kindly check the code snippet below for your peruse
  
```JS
// ./src/redux/books/booksSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer
```

- [ ] Also i would suggest not to leave the `console.log` in your work as it pre-dispose your work to the hackers and this practice is not a best practice of writing JavaScript codes. 

Below is the video illustration of my claim: 👇 

[New Recording - 3_2_2023, 5_24_29 PM.webm](https://user-images.githubusercontent.com/78634841/222609355-a0c9dadf-4af4-4aad-b54e-8beb67398e5c.webm)

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => {
+      if (state.status === false) {
+        state.msg = 'under construction';
+      }
+    },
+  },
```
Comment (brainconnect93) [inline src/redux/categories/categoriesSlice.js:19]: - [ ] As stated in the project required in the categories page there should be a button called `check status` and when the button is been clicked, there should be a text display above or below the button stating `Under Construction`. But currently in your project and i cant see any button in the categories page, rather what i see is a text `Hello Categories`.  Kindly work on this part too

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
## STATUS: Changes Required :recycle:

### Hi @edahigure, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 
-  Has a proper Pull Request. ✔️  
- No Linter errors :heavy_check_mark: 
- Good use of Github flow ✔️ 

### Changes Required

- [ ] In order to make your README file descriptive and more professional, I would kindly advise you work on the MIT license link as currently in your project the link is broke and throwing an error 404.

MIT 👇 

mit

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @edahigure , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**